pkg/actuators/machine/reconciler.go

```diff
@@ -446,3 +474,226 @@ func (r *Reconciler) setMachineAddresses(instance *models.PVMInstance) error {
 	r.machineScope.machine.Status.Addresses = networkAddresses
 	return nil
 }
+
+// updateLoadBalancers adds a given machine instance to the load balancers specified in its provider config
+func (r *Reconciler) updateLoadBalancers(internalIP string) error {
+	if len(r.providerSpec.LoadBalancers) == 0 {
+		klog.V(3).Infof("%s: Instance has no load balancers configured. Skipping", r.machine.Name)
+		return nil
+	}
+	var applicationLoadBalancerNames []string
+	for _, loadBalancerRef := range r.providerSpec.LoadBalancers {
+		switch loadBalancerRef.Type {
+		case machinev1.ApplicationLoadBalancerType:
+			applicationLoadBalancerNames = append(applicationLoadBalancerNames, loadBalancerRef.Name)
+		}
+	}
+	if len(applicationLoadBalancerNames) > 0 {
+		err := registerWithApplicationLoadBalancers(r.powerVSClient, applicationLoadBalancerNames, internalIP)
+		if err != nil {
+			klog.Errorf("%s: Failed to register application load balancers: %v", r.machine.Name, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func registerWithApplicationLoadBalancers(powerVSClient client.Client, loadBalancerNames []string, internalIP string) error {
+	lbMap, err := getLoadBalancers(powerVSClient, loadBalancerNames)
+	if err != nil {
+		return err
+	}
+	return createLoadBalancerPoolMember(powerVSClient, lbMap, internalIP)
+}
+
+func getLoadBalancers(powerVSClient client.Client, loadBalancerNames []string) (map[string]string, error) {
+	// construct the map from provided LoadBalancer names for easy search
+	loadBalancerNamesMap := make(map[string]struct{})
+	for _, lb := range loadBalancerNames {
+		loadBalancerNamesMap[lb] = struct{}{}
+	}
+
+	listLoadBalancersOptions := &vpcv1.ListLoadBalancersOptions{}
+	loadBalancersList, _, err := powerVSClient.ListLoadBalancers(listLoadBalancersOptions)
+	if err != nil {
+		return nil, err
+	}
+	if loadBalancersList == nil {
+		return nil, fmt.Errorf("no loadbalancer is retrieved")
+	}
+
+	// construct LoadBalancer name with LoadBalancer ID map
+	lbMap := make(map[string]string)
+	for _, lb := range loadBalancersList.LoadBalancers {
+		if _, ok := loadBalancerNamesMap[*lb.Name]; ok {
+			lbMap[*lb.Name] = *lb.ID
+		}
+	}
+	if len(lbMap) != len(loadBalancerNames) {
+		return nil, fmt.Errorf("not able to find %s loadbalancer in cloud", loadBalancerNames)
+	}
+	return lbMap, nil
+}
+
+func createLoadBalancerPoolMember(powerVSClient client.Client, lbMap map[string]string, internalIP string) error {
+	// update LoadBalancer pool for each LoadBalancer
+	for lbName, lbID := range lbMap {
+		// Fetch the LoadBalancer details
+		loadBalancer, _, err := powerVSClient.GetLoadBalancer(&vpcv1.GetLoadBalancerOptions{
```

Comment:
Can we see whether we can leverage List call itself to store the Pools and ProvisioningStatus as part of lbMap? I saw List is also returning the same obj that Get returns. so that you can avoid a call here for each LB.